kasa/cli/discover.py

```diff
@@ -201,8 +202,10 @@ def _echo_discovery_info(discovery_info) -> None:
     if discovery_info is None:
         return
 
-    if "system" in discovery_info and "get_sysinfo" in discovery_info["system"]:
-        _echo_dictionary(discovery_info["system"]["get_sysinfo"])
+    if sysinfo := IotDevice.extract_sys_info(discovery_info):
+        if "system" in sysinfo:
+            sysinfo = IotCamera.extract_sys_info(discovery_info)
```

Comment:
Remove `extract_sys_info` from `IotCamera` and extract it out inside `IotDevice`'s implementation.

Reply:
So 1 method to handle both non-camera and camera `sys_info` extraction? [I had that implemented before](https://github.com/python-kasa/python-kasa/pull/1270/commits/e158c4ffaae87dbbef8742a27d6d76c1299fdd95#diff-3ae5c586ca45c6ccd07a881aac8533a04afd1a41ba8749ce99b4f7050a66f7afR440), I can re-add that as a function (not as a method inside `IotDevice`) inside `IotDevice.py`.